cli/azd/pkg/tools/github/github.go

```diff
@@ -517,7 +517,7 @@ func extractFromTar(src, dst string) (string, error) {
 		// cspell: disable-next-line `Typeflag` is comming fron *tar.Header
 		if fileHeader.Typeflag == tar.TypeReg && fileName == "gh" {
 			filePath := filepath.Join(dst, fileName)
-			ghCliFile, err := os.OpenFile(filePath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, os.FileMode(fileHeader.Mode))
+			ghCliFile, err := os.OpenFile(filePath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, fileHeader.FileInfo().Mode())
```

Comment:
This is shifting the problem to the library without actually changing behavior.. Under the covers, the same [conversion](https://cs.opensource.google/go/go/+/refs/tags/go1.23.0:src/archive/tar/common.go;drc=d838e4dcdf89124ed051e1c53e8472e900664a6b;l=563) happens?

Reply:
Can't think of anything better here either.